main/client/src/mill/main/client/FileToStreamTailer.java

```diff
@@ -0,0 +1,97 @@
+package mill.main.client;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.util.Optional;
+
+public class FileToStreamTailer extends Thread implements AutoCloseable {
+
+	private final File file;
+	private final PrintStream stream;
+	private final int intervalMsec;
+
+	private Optional<BufferedReader> reader = Optional.empty();
+	// if true, we won't read the whole file, but only new lines
+	private boolean ignoreHead = true;
+
+	private volatile boolean keepReading = true;
+	private volatile boolean flush = false;
+
+	public FileToStreamTailer(File file, PrintStream stream, int intervalMsec) {
+		super("Tail");
+		this.intervalMsec = intervalMsec;
+		setDaemon(true);
+		this.file = file;
+		this.stream = stream;
+	}
+
+	@Override
+	public void run() {
+		if (isInterrupted()) {
+			keepReading = false;
+		}
+		while (keepReading || flush) {
+			flush = false;
+			try {
+				if (!reader.isPresent()) {
```

Comment:
Honestly I think this might be easier to read as an `if-else` on `null`s; the asymmetry between `if(!reader.isPresent())` and `reader.ifPresent` is pretty awkward

Reply:
This isn't starting an if-else block. Only in case a reader is not present, it will be initialized.

The following `ifPresent` uses the (now always initialized) reader.